server.go

```diff
@@ -657,6 +681,31 @@ func (srv *Server) readUDP(conn *net.UDPConn, timeout time.Duration) ([]byte, *S
 	return m, s, nil
 }
 
+// get shutdown signal if is nil, just make it
+func (srv *Server) getShutdownSignal() chan bool {
+	srv.chanLock.Lock()
+	if srv.shutdownSignal == nil {
+		srv.shutdownSignal = make(chan bool)
```

Comment:
This should move to the server's initialization. Can probably then also get rid of chanLock

Reply:
awesome, testing now